R/fast_outliers.R

```diff
@@ -4,12 +4,99 @@
+        stop("The only permitted values are: 'z-score' or 'interquantile'")
+    }
+ 
+    if (tolower(cols) == 'all'){
+        cols = colnames(data)        
+    }
+    no_nans_list = list()
+    col_type_list = list()
+    perc_nans_list = list()
+    outlier_values_list = list()
+    outlier_count_list = list()
+    method_list = list()
+
+    subset_data = data[cols]
+
+    for (i in cols) {
+        no_nans = sum(is.na(subset_data[[i]]))
+        no_nans_list = append(no_nans_list,no_nans)
+        col_type_list = append(col_type_list,class(subset_data[[i]]))
+        perc_nans_list = append(perc_nans_list,round(no_nans/length(subset_data[[i]]),2))
+        data_no_nans = na.omit(subset_data[[i]])
+        if (class(data_no_nans) %in% c('numeric','integer','double')){
+            if (tolower(method) == "z-score"){
+                score = abs(scale(data_no_nans))
+                outlier_values = list(data_no_nans[which(score>2)])
+                outlier_count_list = append(outlier_count_list,length(outlier_values[[1]]))
+                outlier_values_list = append(outlier_values_list,outlier_values)
+                method_list = append(method_list,"Z-Score")
+                } else if (tolower(method) == "interquantile") {
+                quantiles = quantile(data_no_nans,probs = c(0.25,0.75))
+                Q1 = quantiles[1]
+                Q3 = quantiles[2]
+                IQR = Q3 - Q1
+                score = (data_no_nans < (Q1 - 1.5 * IQR)) | (data_no_nans > (Q3 + 1.5 * IQR))
+                outlier_values = list(data_no_nans[which(score>0)])
+                outlier_count_list = append(outlier_count_list,length(outlier_values[[1]]))
+                outlier_values_list = append(outlier_values_list,outlier_values)
+                method_list = append(method_list,"Interquantile")
+                }
+            } else if (class(data_no_nans) %in% c('character','factor')){
+            score = (table(data_no_nans)/sum(table(data_no_nans)))
+            print(score)
+            outlier_values = list(labels(which(score<threshold_low_freq)))
+            print(outlier_values)
+            print(table(data_no_nans[score<threshold_low_freq]))
+            outlier_count_list = append(outlier_count_list,sum(table(data_no_nans[which(score<threshold_low_freq)])))
+            outlier_values_list = append(outlier_values_list,outlier_values)
+            method_list = append(method_list,"low-freq")
+            }
+    }
+    
+    summary = tibble(
+        column_name = cols,
+        type = as.character(col_type_list),
+        no_nans = as.integer(no_nans_list),
+        perc_nans = as.double(perc_nans_list),
+        outlier_method = method_list,
+        no_outliers = outlier_count_list,
+        outlier_values = outlier_values_list
+    )
+    
+    ##Validating produced summarry contains the same number of requested columns
+
+    if (!(count(summary)[[1]]) == length(cols)){
```

Comment:
```suggestion
    if (!(plyr::count(summary)[[1]]) == length(cols)){
```

Reply:
Hi Subing, I changed the library to "dplyr". I did a test if the function run with that one, and it did. Also when I don ?count() on jupyter lab, thats the one Im being referred to.